phases/19-capstone-projects/49-lm-eval-harness/code/main.py

```
@@ -0,0 +1,493 @@
+    metric = examples[0].metric
+    assert all(ex.metric == metric for ex in examples), f"task {task_name} mixes metrics"
+    metric_fn = METRIC_FNS[metric]
+    per_example: List[Dict[str, object]] = []
+    correct_sum = 0.0
+    total = 0
+    start = time.perf_counter()
+    for i in range(0, len(examples), batch_size):
+        chunk = examples[i:i + batch_size]
+        prompts = [ex.prompt for ex in chunk]
+        outputs = adapter.generate(prompts)
+        for ex, out in zip(chunk, outputs):
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)